### PR TITLE
Add 10 more health connect sensors

### PR DESCRIPTION
**Pull request by dshokouhi**
## Summary

Re-arranges the code to make some things easier to find. I am sure some of the names/descriptions/icons will need to be reconsidered.

Adds:

- Basal metabolic rate
- Body temperature
- Bone mass
- Diastolic blood pressure
- Height
- Lean body mass
- Oxygen saturation
- Respiratory rate
- Systolic blood pressure
- VO2 max

## Screenshots

image

## Link to pull request in Documentation repository

Documentation: home-assistant/companion.home-assistant#1140

## Any other notes

Tested against my own data and also health connect toolbox

### CODE REVIEW

**Comment by dshokouhi** [comment]
needs to be updated after #4830 merges

File: common/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -1316,4 +1316,24 @@
...
+    <string name="basic_sensor_name_blood_pressure_systolic">Blood pressure systolic</string>
+    <string name="basic_sensor_name_body_temperature">Body temperature</string>
+    <string name="sensor_description_body_temperature">Last recorded body temperature from Health Connect</string>
+    <string name="basic_sensor_name_bone_mass">Bone mass</string>
+    <string name="sensor_description_bone_mass">Last recorded bone mass from Health Connect</string>
+    <string name="basic_sensor_name_height">Height</string>
+    <string name="sensor_description_height">Last recorded height from Health Connect</string>
+    <string name="basic_sensor_name_lean_body_mass">Lean body mass</string>
+    <string name="sensor_description_lean_body_mass">Last recorded lean body mass from Health Connect</string>
+    <string name="basic_sensor_name_oxygen_saturation">Oxygen saturation</string>
```
[the remark this answers is not in the document]
Comment (dshokouhi) [reply common/src/main/res/values/strings.xml:1333]: hmm the record itself is [oxygen saturation](https://developer.android.com/reference/kotlin/androidx/health/connect/client/records/OxygenSaturationRecord) 🤔 personally im indifferent here

fitbit says Oxygen Saturation FWIW

image

File: app/src/full/java/io/homeassistant/companion/android/sensors/HealthConnectSensorManager.kt [hunk 2]
```diff
@@ -510,4 +860,55 @@ class HealthConnectSensorManager : SensorManager {
...
+            BloodPressureRecord.BODY_POSITION_LYING_DOWN -> "lying_down"
+            BloodPressureRecord.BODY_POSITION_RECLINING -> "reclining"
+            BloodPressureRecord.BODY_POSITION_SITTING_DOWN -> "sitting_down"
+            BloodPressureRecord.BODY_POSITION_STANDING_UP -> "standing"
+            BloodPressureRecord.BODY_POSITION_UNKNOWN -> STATE_UNKNOWN
+            else -> STATE_UNKNOWN
+        }
+    }
+
+    private fun getMeasurementLocation(location: Int): String {
```
[the remark this answers is not in the document]
Comment (dshokouhi) [reply app/src/full/java/io/homeassistant/companion/android/sensors/HealthConnectSensorManager.kt:875]: ah good call will update

**Comment by jpelgrom** [comment]
There's a merge conflict to resolve

**Pull request merged**